Documentation/sourcebuild-in-repos/adding-features.md

```diff
@@ -0,0 +1,15 @@
+# New features
+
+New features are great!  They also have source-build considerations though.  .NET is no longer just multi-platform but also multi-distribution: there's a Linux SDK produced by Microsoft, but there's also a Linux SDK produced by Red Hat, one produced by Fedora, and one that anyone in the community can build himself from source.
+
+## Things to consider
+
+New features, or expansions of current features, should act sensibly across Windows, Linux, and OSX.  This also involves taking into account the limitations and conventions of the different platforms - for instance, on Linux, it's typical for the .NET SDK to be installed by root and therefore be unchangeable by normal users, so installing global tools needs to take into account the fact that the user might not be able to add anything to the SDK directories.
+
+XXX details on the global tools issues
+
+New features also need to be compatible across all distributions of .NET - for instance, Fedora and Debian cannot distribute *any* non-open-source code, even samples, tests, or minor additions to licenses like restricting the field that the code can be used in.  This includes any code or packages that are used to build product code as well.
```

Comment:
Should we mention that all of our code and dependencies should be MIT licensed? I'm not aware of any exceptions to this.

Reply:
ASP.NET Core is MIT: https://github.com/dotnet/aspnetcore/blob/main/LICENSE.txt

Reply:
If we want to point repo owners to a list of approved licenses, https://opensource.org/licenses/alphabetical seems like a good place.

Fedora has a list of good and bad licenses here: https://fedoraproject.org/wiki/Licensing:Main#Good_Licenses which might be useful as an additional reference.

Reply:
Added references, thanks.